tasks/main.yml

```diff
@@ -668,36 +668,32 @@
           {% endfor %}
           {{ hadr_curr_primary }}
 
-    - name: Configure the {{ mssql_ha_ag_name }} availability group
-      vars:
-        __mssql_input_sql_file: configure_ag.j2
-      include_tasks: input_sql_file.yml
+    - name: Configure the current primary replica
       when: >-
         (__mssql_ha_curr_primary | length == 1
         and
-        __mssql_sqlcmd_input_file.stdout_lines[2] | int == 1
+        __mssql_ha_curr_primary[0] == ansible_hostname
         )
         or
         (__mssql_ha_curr_primary | length == 0
         and
         mssql_ha_replica_type == 'primary')
-
-- name: Finish configuration on the primary replica
-  any_errors_fatal: true
-  when:
-    - mssql_ha_configure | bool
-    - mssql_ha_replica_type == 'primary'
-  block:
-    - name: Grant permissions to the {{ mssql_ha_login }} login
-      vars:
-        __mssql_input_sql_file: grant_permissions_to_ha_login.j2
-      include_tasks: input_sql_file.yml
-
-    - name: Back up and replicate databases
-      vars:
-        __mssql_input_sql_file: replicate_db.j2
-      include_tasks: input_sql_file.yml
-      when: mssql_ha_db_names is not none
+      block:
+        - name: Configure the {{ mssql_ha_ag_name }} availability group
+          vars:
+            __mssql_input_sql_file: configure_ag.j2
+          include_tasks: input_sql_file.yml
+
+        - name: Grant permissions to the {{ mssql_ha_login }} login
+          vars:
+            __mssql_input_sql_file: grant_permissions_to_ha_login.j2
+          include_tasks: input_sql_file.yml
+
+        - name: Back up and replicate databases
+          vars:
+            __mssql_input_sql_file: replicate_db.j2
+          include_tasks: input_sql_file.yml
+          when: mssql_ha_db_names is not none
```

Comment:
I guess this will change when https://github.com/linux-system-roles/mssql/pull/116 is merged